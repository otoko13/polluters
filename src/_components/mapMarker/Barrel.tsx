import React from 'react';
import { ReactComponent as BarrelSvg } from '../../resources/barrel.svg';
import './barrel.scss';

export enum EHighlightStatus {
  Highlighted,
  unhighlighted,
  Neutral,
}

export interface IBarrelProps {
  rank: number;
  status: EHighlightStatus;
}

const Barrel = (props: IBarrelProps) => {
  function getHighlightClass() {
    switch (props.status) {
      case EHighlightStatus.Highlighted:
        return 'highlighted';
      case EHighlightStatus.unhighlighted:
        return 'unhighlighted';
      default:
        return '';
    }
  }

  return (
    <g className={`Barrel ${getHighlightClass()}`}>
      <BarrelSvg />
    </g>
  );
};

export default Barrel;
