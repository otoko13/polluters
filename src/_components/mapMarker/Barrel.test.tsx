import { render } from '@testing-library/react';
import * as React from 'react';
import HomeIcon from '@material-ui/icons/Home';
import Barrel, { IBarrelProps, EHighlightStatus } from './Barrel';

let props: IBarrelProps;

beforeEach(() => {
  props = {
    magnification: 1,
    rank: 1,
    status: EHighlightStatus.Neutral,
  };
});

describe('snapshots', () => {
  it('should look sensible', () => {
    const { container } = render(<Barrel {...props} />);
    expect(container.firstChild).toMatchSnapshot('State');
  });
});
