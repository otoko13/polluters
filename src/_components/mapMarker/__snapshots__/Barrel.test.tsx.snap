// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should look sensible 1`] = `
<svg>
  <g
    class="Barrel  rank-1"
  >
    <g
      style="transform: scale3d(1, 1, 1);"
    >
      <g
        class="barrel-content"
      >
        <svg>
          barrel.svg
        </svg>
      </g>
    </g>
  </g>
</svg>
`;
